sktime/clustering/_k_shapes.py

```diff
@@ -0,0 +1,154 @@
+# -*- coding: utf-8 -*-
+from typing import Union
+
+import numpy as np
+from numpy.random import RandomState
+
+from sktime.clustering._base import BaseClusterer, TimeSeriesInstances
+from sktime.utils.validation._dependencies import _check_soft_dependencies
+
+_check_soft_dependencies("tslearn")
```

Comment:
@chrisholder I'm not 100% sure since readthedocs logs are not very transparent but I think might be triggering the error when this module is being imported in  `sktime/clustering/__init__.py`

Reply:
Maybe doing the ckeck inside class/init?

Reply:
That seems like a good option @aiwalter! If we want to follow the `sklearn` doctrine it might be better to put in `fit` to avoid side effects but it is also late in the execution sequence. As a user I would like to get the soft dependency info as fast as possible which practically means on importing the class from the module. One way of resolving this might be to update `_check_soft_dependencies` to act differently when in building the docs by using the env var that the rtfd sets during build https://docs.readthedocs.io/en/stable/faq.html#how-do-i-change-behavior-when-building-with-read-the-docs That would allow for documenting all the modules with soft dependencies and not actually require them to build the docs.

Reply:
It cant be in `fit` as we need to instantiate the dependency usually in `init`, at least in the forecasters

Reply:
So if Im reading correctly here: for now if I remove _k_shapes from the init, and then longer term we can put in place a solution that will allow us to have soft dependency files inside init.

Reply:
@lmmentel, would you be so kind and open a "good first issue" that outlines your proposed solution to the soft dependency handling, so someone from the community could pick it up?